### DISCUSSION TITLE
How do "Time of Use" working mode and "Charge from Grid" interact with each other?

### DISCUSSION THREAD

**Discussion by jodewee** [wlcrs/huawei_solar#132] [General]
### System Health details

The charge from grid-option seems not be working;

![image](https://user-images.githubusercontent.com/40162805/189073597-b8c46a8f-f78f-422a-bb33-2b720f4492ba.png)

### Huawei Solar Setup

SUN2000-4.6KTL-L1 with 5kw LUNA 2000

### Describe the issue

when switching "charge from grid" and set the mode to: time of use;
the battery isn't charging from the grid

### Relevant debug logs

```text
/
```

**Comment by wlcrs** [comment]
I've moved this to a discussion, as this is not a bug but a configuration problem. I'm not completely sure how the "Charge from grid" toggle works in practice. I've only tested it with the "forcible charge" functionality.

Maybe you can share how you have configured the Time of Use timeslots on your inverter, so that others can help?

**Comment by jodewee** [comment]
ok, that's probably the issue; i did not configure anything on timeslots on the inverter.
